kahuna/public/js/components/gr-image-usage-photosales/gr-image-usage-photosales.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from "react";
+import * as angular from "angular";
+import { react2angular } from "react2angular";
+import {useEffect, useState} from "react";
+import * as moment from 'moment';
+import "./gr-image-usage-photosales.css";
+
+export interface Usage {
+  title: string,
+  usageName: string,
+  usageType: string,
+  dateAdded: Date
+}
+
+export interface UsageProps {
+  usages: Usage[];
+}
+
+export interface UsagePropsWrapper {
+  props: UsageProps;
+}
+
+const ImageUsagePhotoSales: React.FC<UsagePropsWrapper> = ({ props }) => {
+  const [usages, setUsages] = useState(props.usages);
+  const [extractedPhotoSalesUsage, setPhotoSalesUsage] = useState<Usage|null>(null);
+
+  const photoSalesIdentifier = "sendToPhotoSales";
+
+  const formatTimestamp = (timestamp: Date) => {
+    return moment(timestamp).fromNow();
+  };
+
+  useEffect(()=> {
```

Comment:
You can remove this `useEffect` block by inlining the initialisation of the state. Ie:

```
const [photoSalesUsage] = useState<Usage | undefined>(usages().find(usage => usage.usageType === "sendToPhotoSales"));
```

This highlighted that the UsageProps interface is actually incorrect here. It should be:

```
export interface UsageProps {
  usages: () => Usage[];
}
```

You are passing in a function, not an array.